### ISSUE TITLE
image too large, max allowed 8192 kB

### ISSUE THREAD

**Issue by silviu25** [PrestaShop/PrestaShop#15183]
>Edited by Khouloud

In the BO => Advanced Parameters => Administration we have an option "Maximum size for a product's image" but has no effect.
I manage to reproduce the issue with PS1.7.6.0 & PS1.7.5.2.
`Maximum size for a product's image` has no effect on the `PS_PRODUCT_PICTURE_MAX_SIZE` on the database.
We have weird behavior
In the FO / BO, when we try to upload a large image => NOK
When we try to add the same image in the description section (Cms page, Product) => OK
![image](https://user-images.githubusercontent.com/16067358/63603083-c1f59280-c5c0-11e9-9351-84756da554c1.png)
It matches the default php configuration for post_max_size which is 8 mb
https://www.php.net/manual/en/ini.core.php
![image](https://user-images.githubusercontent.com/16067358/63603192-03863d80-c5c1-11e9-8971-21e6642f396b.png)

------------------------------------------
**Describe the bug** **To Reproduce**
I try to upload a image for a product which has customizations bt I get the following error: "Image is too large (48231 kB). Maximum allowed: 8192 kB"

nowhere on the server or in prestashop do I have this limit of 8192 kB or 8mb

In php I have upload_max_filesize = 512M and post_max_size = 512M
In Admin>Advanced parameters>Upload quota I have: The maximum size for the attached files 512 and Maximum image size of a product 512

**Screenshots**
https://ibb.co/r0zr02g
https://ibb.co/4MDq8B5
https://ibb.co/TvbNdHH

**Additional information**
PrestaShop version: 1.7.6.0
PHP version: 7.2.21

**Comment by khouloudbelguith** [comment]
Hi @silviu25,

"1G" is not a valid maximum size
![image](https://user-images.githubusercontent.com/16067358/63425776-3d144880-c409-11e9-9ea2-14a03ad3b826.png)
Thanks to change it to `1000M` for example.
unfortunately, we are aware of this issue and it does not come from Prestashop, see https://github.com/PrestaShop/PrestaShop/pull/10892#issuecomment-439097520

Thanks!

**Comment by khouloudbelguith** [comment]
Duplicate of #15162

**Comment by silviu25** [comment]
that is memory limit not maximum size

**Comment by khouloudbelguith** [comment]
@silviu25, Could you please follow this link: https://github.com/PrestaShop/php-ps-info
phppsinfo file provides an equivalent to the phpinfo() function that reports PrestaShop Requirements information about the PHP/MySQL/Apache environment and offers suggestions for improvement.

1. Upload the file to the server. You should upload your file to the exact directory you want to test. Typically, this will be your httpdocs (/var/www/vhosts/example.com/httpdocs/) directory, although you can upload it to any subdirectory on your server as well. Use FTP to upload the file.
2. Visit the page in your browser. If you uploaded it to your html directory, you should now visit http://www.example.com/phppsinfo.php, replacing example.com with your own domain name.
3. Now you can view all of the information about PHP for your server for that particular directory.

![image](https://user-images.githubusercontent.com/16067358/44723217-a8199080-aacf-11e8-8c33-171ecbf3b5db.png)

**Comment by silviu25** [comment]
<a href="https://ibb.co/CP70zTv"><img src="https://i.ibb.co/sWw32MQ/x.png" alt="x" border="0"></a>

**Comment by khouloudbelguith** [comment]
@silviu25, have you tried to change the memory limit & check if the issue occurs again?
![image](https://user-images.githubusercontent.com/16067358/63427637-81094c80-c40d-11e9-8cda-8aaa64f5c712.png)
Thanks!

**Comment by silviu25** [comment]
I changed to 769M but still no luck

<a href="https://ibb.co/VYpyTT1"><img src="https://i.ibb.co/F4qvzzr/xx.png" alt="xx" border="0"></a>

**Comment by silviu25** [comment]
any idea?

**Comment by khouloudbelguith** [comment]
@silviu25, could you please navigate to your BO => Catalog => Files => Add new file => upload your image, check & feedback.

Thanks!

**Comment by silviu25** [comment]
It is working in BO

<a href="https://ibb.co/ggpy94g"><img src="https://i.ibb.co/WKwn5VK/8.jpg" alt="8" border="0"></a>

**Comment by khouloudbelguith** [comment]
@silviu25, if it is an FO issue only, could you please upload this file in the "contact form" page & check if you have the same error?
![image](https://user-images.githubusercontent.com/16067358/63431909-e06c5a00-c417-11e9-8b8d-8983eacc8231.png)


Thanks!

**Comment by silviu25** [comment]
I receive this duplicated message: "There was an error sending the message."
<a href="https://ibb.co/7rfbTDB"><img src="https://i.ibb.co/VM13hGb/90.jpg" alt="90" border="0"></a>

**Comment by silviu25** [comment]
but without attached file is working ok

**Comment by silviu25** [comment]
even so I received the message with attachment in BO > Customer support

**Comment by khouloudbelguith** [comment]
@silviu25, no this a different issue.
What is the exact type of this product? is it a standard product?

Thanks!

**Comment by silviu25** [comment]
Is a product with combinations
I'm selling canvas prints and the customer choose the size and material then upload his picture. the price for this product is updated according to material and size. 
Here you can test: shorturl.at/gIV67

**Comment by khouloudbelguith** [comment]
@silviu25, I tried with this image in your shop
![legumes-clairemedium](https://user-images.githubusercontent.com/16067358/63434114-5b377400-c41c-11e9-9c07-e47e7392e43b.jpg)
& it is well uploaded.
![image](https://user-images.githubusercontent.com/16067358/63434148-6c808080-c41c-11e9-80f9-bc42dddef72e.png)

Thanks!

**Comment by silviu25** [comment]
is ok because is small 445 KB, is under 8192 kB
what I am trying to do is to increase this limit of 8192 kB
clients need to upload 45,000 KB files or even larger

**Comment by khouloudbelguith** [comment]
@silviu25, thanks!
So, can I close the issue?

Thanks!

**Comment by silviu25** [comment]
no, we still can not upload large images because for some reason prestashop have a 8192 kB upload limit

**Comment by silviu25** [comment]
I found this: https://www.prestashop.com/forums/topic/355339-maximum-size-allowed/?tab=comments#comment-1896136 but but it does not appear to be applicable to PS 1.7.6.0

**Comment by khouloudbelguith** [comment]
@silviu25, Could you please provide me the exact image with the large size to test it.
This is a public space.
You can provide me the picture by email.
My address mail: khouloud.belguith@prestashop.com
Let's be safe!

Thanks!

**Comment by silviu25** [comment]
Here are some free large images you can test https://www.effigis.com/en/solutions/satellite-images/satellite-image-samples/ 
and by email you have the picture in question
Thanks

**Comment by khouloudbelguith** [comment]
@silviu25, yes!
the `PS_PRODUCT_PICTURE_MAX_SIZE` is limited to `8388608`
https://github.com/PrestaShop/PrestaShop/blob/develop/install-dev/data/xml/configuration.xml#L105
In your database ps_configuration table, you can check it
![image](https://user-images.githubusercontent.com/16067358/63591626-34f20f80-c5a7-11e9-93a6-8fa518fc733d.png)

Thanks!

**Comment by silviu25** [comment]
Thank you, I will modify and come back with results later.

But this shouldn't be editable from BO?

**Comment by khouloudbelguith** [comment]
@silviu25, In the BO => Advanced Parameters => Administration we have an option "Maximum size for a product's image" but has no effect.
I manage to reproduce the issue with PS1.7.6.0 & PS1.7.5.2.
`Maximum size for a product's image` has no effect on the `PS_PRODUCT_PICTURE_MAX_SIZE` on the database.
We have weird behavior
In the FO / BO, when we try to upload a large image => NOK
When we try to add the same image in the description section (Cms page, Product) => OK
![image](https://user-images.githubusercontent.com/16067358/63603083-c1f59280-c5c0-11e9-9351-84756da554c1.png)
It matches the default php configuration for post_max_size which is 8 mb
https://www.php.net/manual/en/ini.core.php
![image](https://user-images.githubusercontent.com/16067358/63603192-03863d80-c5c1-11e9-8971-21e6642f396b.png)


I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by silviu25** [comment]
I changed the value of `PS_PRODUCT_PICTURE_MAX_SIZE` in `ps_configuration` from `8388608` to '52428800"  [from 8mb to 50mb] and In the BO => Advanced Parameters => Administration > Maximum size for a product's image to 50 and now is working

but there is another problem:

after I uploaded picture for the product and I placed the order, I received that order in BO
then I go to Orders>Orders>down to "Products" field and I downloaded the picture previously uploaded as a customer

<a href="https://ibb.co/fD0Zqdz"><img src="https://i.ibb.co/DrKmzfH/Untitled.jpg" alt="Untitled" border="0"></a>

and the downloaded image is smaller in size, it was reduced from 39,1mb to 11,2mb 
<a href="https://ibb.co/4KRxH89"><img src="https://i.ibb.co/Nt2zM9Q/Untitled.jpg" alt="Untitled" border="0"></a>

and I'm not using any "Image Compress or optimization" module

Shouldn't the image uploaded by the buyer have the same size as the image I download from the BO as a seller?

**Comment by khouloudbelguith** [comment]
@silviu25, I tried with PS1.7.6.0 & the size image is not descreased
![image](https://user-images.githubusercontent.com/16067358/63698801-843e7700-c817-11e9-9e31-60f1cbc6f717.png)

Thanks!

**Comment by Hlavtox** [comment]
Closing as the old product page is gone now.
